app/services/meta-tags.ts

```diff
@@ -0,0 +1,184 @@
+import { service } from '@ember-decorators/service';
+import Service from '@ember/service';
+import config from 'ember-get-config';
+import pathJoin from 'ember-osf-web/utils/path-join';
+
+export type NameAttr = 'name' | 'property';
+
+export type Content = string | number | null | undefined;
+
+export type DataContent = Content | Content[];
+
+/* eslint-disable camelcase */
+export interface MetaTagsData {
+    title: DataContent;
+    type?: DataContent;
+    description?: DataContent;
+    url?: DataContent;
+    doi?: DataContent;
+    identifier?: DataContent;
+    published_date?: DataContent;
+    modified_date?: DataContent;
+    license?: DataContent;
+    language?: DataContent;
+    image?: DataContent;
+    image_type?: DataContent;
+    image_width?: DataContent;
+    image_height?: DataContent;
+    image_alt?: DataContent;
+    site_name?: DataContent;
+    institution?: DataContent;
+    twitter_site?: DataContent;
+    twitter_creator?: DataContent;
+}
+/* eslint-enable camelcase */
+
+export interface MetaTagsDefs {
+    [s: string]: DataContent;
+}
+
+export interface NameMetaTagAttrs {
+    name: string; // eslint-disable-line no-restricted-globals
+    content: Content;
+}
+
+export interface PropMetaTagAttrs {
+    property: string;
+    content: Content;
+}
+
+export type MetaTagAttrs = NameMetaTagAttrs | PropMetaTagAttrs;
+
+export interface HeadTagDef {
+    type: string;
+    attrs: MetaTagAttrs;
+}
+
+export default class MetaTags extends Service {
+    @service i18n;
+
+    /**
+     * Get meta tag definitions.
+     *
+     * @method getMetaTags
+     * @param {MetaTagsData} metaTagsOverrides Data values to override defaults.
+     * @return {MetaTagsDefs} Returns meta tag definitions.
+     */
+    getMetaTags(this: MetaTags, metaTagsOverrides: MetaTagsData): MetaTagsDefs {
+        // Default values.
+        const metaTagsData: MetaTagsData = {
+            title: undefined,
+            type: 'collection',
+            description: this.get('i18n').t('general.hosted_on_the_osf'),
+            url: window.location.href,
+            doi: undefined,
+            identifier: undefined,
+            published_date: undefined,
+            modified_date: undefined,
+            license: undefined,
+            language: 'en',
+            image: pathJoin(config.OSF.url, 'static/img/preprints_assets/osf/sharing.png'),
+            image_type: 'image/png',
+            image_width: 1200,
+            image_height: 630,
+            image_alt: this.get('i18n').t('home.brand'),
+            site_name: this.get('i18n').t('home.brand'),
+            institution: this.get('i18n').t('general.cos'),
+            twitter_site: '@OSFramework',
+            twitter_creator: '@OSFramework',
+        };
+
+        if (metaTagsOverrides === Object(metaTagsOverrides)) {
```

Comment:
The problem is that TypeScript only works if you type things appropriately. If you don't type the injected meta-tags service (either explicitly or via the registry) there will be no type checking on the metaTagsOverrides argument. Perhaps if we had a linting rule that required types for injected services...

That being said, this really doesn't so anything useful as it will just fail silently if you pass in a non-object.

Reply:
Oh, good point. Maybe if we get to the point we can flip on `noImplicitAny`...